src/image.cpp

```diff
@@ -390,8 +390,8 @@ void Image::printIFDStructure(BasicIo& io, std::ostream& out, Exiv2::PrintStruct
       if (allocate64 > io.size()) {
         throw Error(ErrorCode::kerInvalidMalloc);
       }
-      DataBuf buf(allocate64);                     // allocate a buffer
-      std::copy_n(dir.c_data(8), 4, buf.begin());  // copy dir[8:11] into buffer (short strings)
+      DataBuf buf(allocate64);                                       // allocate a buffer
+      std::copy(dir.begin() + 8, dir.begin() + 8 + 4, buf.begin());  // copy dir[8:11] into buffer (short strings)
```

Comment:
Why is this better? It seems more verbose than the old version that used `std::copy_n`.

Reply:
Reworked.